.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,46 @@
+name: Benchmark
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install juliaup
+      uses: julia-actions/install-juliaup@v2.1.2
+      with:
+        channel: '1'
+    - name: Update Julia registry 
+      shell: julia --project=. --color=yes {0}
+      run: |
+        using Pkg
+        Pkg.Registry.update()
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: 3.9
+    - name: Install dependencies
+      run: |
+        pip install pytest pytest-benchmark pytest-xdist juliacall
+        pip install -e . # to put juliapkg.json in sys.path
```

Comment:
why not just `pip install -e .[test]`? This should pull in all of the packages you list?

Reply:
Because I didn't know about this short form ;)